fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryCoveringIndexPlan.java

```diff
@@ -73,19 +75,25 @@ public RecordQueryCoveringIndexPlan(@Nonnull RecordQueryPlanWithIndex indexPlan,
 
     @Nonnull
     @Override
-    @SuppressWarnings("unchecked")
     public <M extends Message> RecordCursor<FDBQueriedRecord<M>> execute(@Nonnull FDBRecordStoreBase<M> store,
                                                                          @Nonnull EvaluationContext context,
                                                                          @Nullable byte[] continuation,
                                                                          @Nonnull ExecuteProperties executeProperties) {
+        return indexPlan
+                .executeEntries(store, context, continuation, executeProperties)
+                .map(indexEntryToQueriedRecord(store));
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    @API(API.Status.INTERNAL)
+    public <M extends Message> Function<IndexEntry, FDBQueriedRecord<M>> indexEntryToQueriedRecord(final @Nonnull FDBRecordStoreBase<M> store) {
         final RecordMetaData metaData = store.getRecordMetaData();
```

Comment:
all these getters are now done per record and could be done once per execute(). Not sure if in one of these you have anything computationally expensive going on...

Reply:
It doesn't map calls to `indexEntryToQueriedRecord` over them; it maps the _result_ of `indexEntryToQueriedRecord`, which is a function, closing over all those variables, over them.

Reply:
ah my bad